src/controllers/UsersController.php

```diff
@@ -298,4 +298,36 @@ public function passwordReset(Request $request, $db)
         }
 
     }
+
+
+
+    public function deleteAction($request, $db)
+    {
+
+
+        if (! isset($request->user_id)) {
+            throw new Exception("You must be logged in to delete data", 400);
+        }
+        // delete the talk
+        $user_id     = $this->getItemId($request);
+        $user_mapper = new UserMapper($db, $request);
+
+        // note: use the mapper's getUserById as we don't want to throw a not found exception
+        $user = $user_mapper->getUserById($user_id);
+        if (false === $user) {
+            // user isn't there so it's as good as deleted
+            header("Content-Length: 0", null, 204);
```

Comment:
This is a somewhat philosophical thing. In my eyes (and I seem not to be the only person - http://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete#comment-10798415) a call to delete a non-existing resource is not a valid request.

Besides that, there should be room for improvement. But I'm with you, that we should be consistent and even if that's consistently strange :wink: